app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,81 @@
+class PasswordResetsController < ApplicationController
+  before_action :take_user, only: %i(edit update)
+  before_action :valid_user, only: %i(edit update)
+  before_action :check_expiration, only: %i(edit update)
+
+  def new; end
+
+  def create
+    user = User.find_by email: params[:password_reset][:email].downcase
+    if user
+      send_mail_valid_user user
+    else
+      flash.now[:danger] = t "password_resets.controller.flash.not_found_email"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      user.errors.add password: t("password_resets.controller.error.pass_empty")
+      render_edit
+    elsif user.update_attributes user_params
+      after_update_pass user
+    else
+      render_edit
+    end
+  end
+
+  private
+
+  attr_reader :user
+
+  def after_update_pass user
+    log_in user
+    flash[:success] = t "password_resets.controller.flash.has_reset_pass"
+    redirect_to user
+  end
+
+  def check_expiration
+    return unless user.password_reset_expired?
+    flash[:danger] = t "password_resets.controller.flash.pass_reset_expired"
+    redirect_to new_password_reset_url
+  end
+
+  def render_edit
+    render :edit
+  end
+
+  def redirect_root
+    redirect_to root_url
+  end
+
+  def take_user
```

Comment:
get_user

Reply:
dùng get_user robocop báo lỗi Do not prefix reader method names with `get_`

Reply:
nhầm, xin lỗi c, t viết nhầm, tên hàm là `find_user` nhé